klusta_pipeline/openephys_to_kwd.py

```diff
@@ -0,0 +1,217 @@
+    
+    # Check if file is consistent
+    if n_record_bytes % RECORD_SIZE != 0:
+        raise Exception("File size inconsistent: possible corrupt data file")
+
+    n_records = n_record_bytes // RECORD_SIZE
+    n_samples = n_records * SAMPLES_PER_RECORD
+    return (n_records, n_samples)
+
+def load_openephys_continuous_record(oe_file, record_num):
+    # Loads a single record from an openephys continous file
+    
+    # Calculate start of record in bytes
+    record_start = 1024 + record_num*RECORD_SIZE
+    # Move to start of record
+    oe_file.seek(record_start)
+    # Get Timestamp
+    record_timestamp = np.fromfile(oe_file, np.dtype('<i8'), 1)
+    # Get Number of samples
+    record_n_samples = np.fromfile(oe_file, np.dtype('<u2'), 1)[0]
+    if record_n_samples != SAMPLES_PER_RECORD:
+        raise Exception('Found corrupted record in ' + str(record_num))
+    # Get recording number
+    record_recording_number = np.fromfile(oe_file, np.dtype('>u2'), 1)[0]
+    # Get raw record data
+    record_raw_data = np.fromfile(oe_file, np.dtype('>i2'), record_n_samples)
+    # Ignore record marker
+    oe_file.read(10)  
+    # Make array of timesamples
+    sample_numbers = np.arange(record_timestamp, 
+                               record_timestamp + record_n_samples)
+    # Make data array
+    data = np.zeros((record_n_samples, 2))
+    data[:, 0] = sample_numbers
+    data[:, 1] = record_raw_data
+    recording_start_sample = int(data[0, 0])
+
+    return (record_num, record_recording_number, recording_start_sample, data)
+
+def get_openephys_continuous_recording_numbers(oe_file):
+    # Returns an array of recording numbers within a .continuous file
+    recordings = []
+    (n_records, n_samples) = calculate_openephys_continuous_sizes(oe_file)
+    for record in range(n_records):
+        record_offset = 1024 + record*RECORD_SIZE
+        recording_offset = record_offset + RECORD_RECORDING_OFFSET
+        oe_file.seek(recording_offset)
+        record_recording_number = np.fromfile(oe_file, np.dtype('>u2'), 1)[0]
+        recordings.append(record_recording_number)
+    return recordings
+
+def get_openephys_continuous_timestamps(oe_file):
+    # Returns an array of recording numbers within a .continuous file
+    timestamps = []
+    (n_records, n_samples) = calculate_openephys_continuous_sizes(oe_file)
+    for record in range(n_records):
+        record_offset = 1024 + record*RECORD_SIZE
+        timestamp_offset = record_offset + RECORD_TIMESTAMP_OFFSET
+        oe_file.seek(timestamp_offset)
+        record_timestamp = np.fromfile(oe_file, np.dtype('<i8'), 1)
+        timestamps.append(record_timestamp)
+    return timestamps
+
+def get_openephys_continuous_record_metadata(oe_file):
+    # Returns an array of recording numbers within a .continuous file
+    recordings = []
+    timestamps = []
+    (n_records, n_samples) = calculate_openephys_continuous_sizes(oe_file)
+    for record in range(n_records):
```

Comment:
this record/record offset code is a bit of a repeated motif throughout several functions, it might be clearer to say 
```
for record_offset in gen_record_offsets(n_records):
    timestamps.append(seek_and_read(oe_file, record_offset + RECORD_TIMESTAMP_OFFSET, '<i8'))
    recordings.append(seek_and_read(oe_file, record_offset + RECORD_RECORDING_OFFSET, '>u2')[0])
```
but that's just my style